### PR TITLE
fix(dropdown): use teleport

### PR DESCRIPTION
**Pull request by nandi95**
This prevents the dropdown to be pushing the parent dimensions triggering possible scroll

### CODE REVIEW

File: src/components/dropdown/UIDropdown.vue [hunk 1]
```diff
@@ -7,33 +7,35 @@
...
 
-        <template v-if="transitionComponent">
-            <component :is="transitionComponent">
-                <div v-if="isOpen"
-                     v-bind="$attrs"
-                     ref="dropdown"
-                     v-click-away="hide"
-                     class="dropdown absolute z-50 rounded overflow-auto flex flex-col items-stretch shadow-lg
+        <Teleport to="body">
+            <template v-if="transitionComponent">
```
Comment (fullstackfool) [inline src/components/dropdown/UIDropdown.vue:11]: Should be able to move this conditional to the `component` line below, tidy up the HTML a little.

File: src/components/dropdown/UIDropdown.vue [hunk 2]
```diff
@@ -7,33 +7,35 @@
...
+                         @click.stop>
+                        <slot :toggle="toggle" :show="show" :hide="hide" />
+                    </div>
+                </component>
+            </template>
+            <div v-else-if="isOpen"
+                 v-bind="$attrs"
+                 ref="dropdown"
+                 v-click-away="hide"
+                 class="dropdown absolute z-50 rounded overflow-auto flex flex-col items-stretch shadow-lg
```
Comment (fullstackfool) [inline src/components/dropdown/UIDropdown.vue:31]: `h-max` missing here, intentional?

File: src/components/dropdown/UIDropdown.vue [hunk 3]
```diff
@@ -151,34 +153,51 @@ export default defineComponent({
...
-                style.top = `${contentParameters.height}px`;
+                let top = globalThis?.document.documentElement.scrollTop
+                    + dropdownRect.top
+                    + dropdownRect.height;
+
+                if (globalThis?.document.documentElement.scrollHeight <= top + contentHeight) {
+                    top = globalThis?.document.documentElement.scrollHeight - contentHeight;
+                }
+
+                style.top = `${top}px`;
```
Comment (fullstackfool) [inline src/components/dropdown/UIDropdown.vue:200]: Regression here: menu no longer aligns itself with the parent when using `vertical="top"` (image shows right-click menu).
 
image

**Review by fullstackfool — CHANGES_REQUESTED** [review]
1 bug to be sorted, rest are questions/suggestions 👍

File: src/components/dropdown/UIDropdown.vue (hunk 1 above)
Comment (nandi95) [reply src/components/dropdown/UIDropdown.vue:11]: If I remember correctly I could not and there was a reason why i hed the `<template>` tag.
I think the `<component>` needs to have it's dependency (TransitionComponent) known when mounting if that makes sense.

**Review by fullstackfool — APPROVED** [review]

File: src/components/dropdown/UIDropdown.vue (hunk 1 above)
Comment (nandi95) [reply src/components/dropdown/UIDropdown.vue:11]: Doesn't seem to be needed anymore or I'm not testing some edge case. Removing.

**Pull request merged**